### PR TITLE
chore: update `Timer.md` with caveats

### PR DESCRIPTION
**Pull request by ggreif**
... and suggestions. Touch on security and the default mechanism.

Also linking @dfinity-berestovskyy's timer FAQ, which is https://internetcomputer.org/docs/current/developer-docs/backend/periodic-tasks#timers-library-limitations

### CODE REVIEW

File: doc/md/base/Timer.md [hunk 1]
```diff
@@ -1,11 +1,25 @@
...
+Timers for one-off or periodic tasks. Applicable as part of the default mechanism.
 
 Note: If `moc` is invoked with `-no-timer`, the importing will fail.
+
 Note: The resolution of the timers is in the order of the block rate,
       so durations should be chosen well above that. For frequent
       canister wake-ups the heatbeat mechanism should be considered.
 
+Note: Below functionality is active only if the user doesn't explicitly
+      implement `system func timer`.
```
Comment (crusso) [inline doc/md/base/Timer.md:12]:
```suggestion

Note: The functionality described below is enabled only when the actor does not override it by declaring an explicit `system func timer`.
```

File: doc/md/base/Timer.md [hunk 2]
```diff
@@ -1,11 +1,25 @@
 # Timer
-Timers for one-off or periodic tasks.
+
+Timers for one-off or periodic tasks. Applicable as part of the default mechanism.
 
 Note: If `moc` is invoked with `-no-timer`, the importing will fail.
+
 Note: The resolution of the timers is in the order of the block rate,
       so durations should be chosen well above that. For frequent
       canister wake-ups the heatbeat mechanism should be considered.
```
Comment (crusso) [inline doc/md/base/Timer.md:9]:
```suggestion
      canister wake-ups the heartbeat mechanism should be considered.
```

**Review by crusso — APPROVED** [review]
But see suggestions

File: doc/md/base/Timer.md [hunk 3]
```diff
@@ -1,11 +1,25 @@
...
 
+Note: Below functionality is active only if the user doesn't explicitly
+      implement `system func timer`.
+
+Note: Timers are _not_ persisted across upgrades. One possible strategy
+      to re-establish timers after an upgrade is to walk stable variables
+      in the `post_upgrade` hook and distill necessary timer information
+      from there.
```
[the remark this answers is not in the document]
Comment (roelstorms) [reply doc/md/base/Timer.md:17]: Is this correct?
> is to **walk** stable variables

File: doc/md/base/Timer.md [hunk 4]
```diff
@@ -1,11 +1,25 @@
...
+
+Note: Basing security (e.g. access control) on timers is almost always
```
Comment (roelstorms) [inline doc/md/base/Timer.md:19]: What's the underlying reason that security controls shouldn't use timers?

I can't come up with a good scenario in which one would want to. Just wondering if this statements mean that timers are not very reliable or that usually if you need timers in a security control you there is a design issue. Agreed that access control would likely not need timers.

Comment (ggreif) [reply doc/md/base/Timer.md:19]: It is triggered by your slack comment. There are access control methods that use time, e.g. exponential lockout for mistyped passwords etc. These are problematic here, as there is no protection against reentrancy.

File: doc/md/base/Timer.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ggreif) [reply doc/md/base/Timer.md:17]: Yep, I think it is common usage to _walk_ (i.e. traverse) data structures.

File: doc/md/base/Timer.md (hunk 4 above)
Comment (ggreif) [reply doc/md/base/Timer.md:18]: I have added a blurb. PTAL.

**Comment by luc-blaeser** [comment]
Just doing some repo cleanup for better focussing on active work:
Closing PR as it seems inactive for a longer time. Please re-open if it is still relevant and after it has been updated with latest master changes.

File: doc/md/base/Timer.md (hunk 4 above)
Comment (roelstorms) [reply doc/md/base/Timer.md:18]: I missed this comment back then. LGTM if you still want to merge it.

**Review by roelstorms — APPROVED** [review]
LGTM, thanks for improving the docs!

**Pull request merged**